packages/react-native-worklets/src/shareables.ts

```diff
@@ -253,6 +253,36 @@ function cloneArray<T extends unknown[]>(
   shouldPersistRemote: boolean,
   depth: number
 ): ShareableRef<T> {
+  if (value.length > 20) {
+    const isArrayOfStrings = value.every(
+      (element) => typeof element === 'string'
+    );
+    const isArrayOfNumbers = value.every(
+      (element) => typeof element === 'number'
+    );
```

Comment:
Let's move this below `if (isArrayOfStrings) { ... }` so that we don't need to evaluate `isArrayOfNumbers`.